Firestore/Swift/Source/ReferenceableObject/FirestoreObjRefWrapper.swift

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import FirebaseFirestore
+
+/// Property wrapper @FirestoreObjectReference,
+/// Indicates that the specified property value should be stored by reference instead of by value,  inline
+/// with the parent object.
```

Comment:
I am not sure what "inline with the parent object" refers to.

Reply:
I can't find a good way to describe this. Checked both these videos but not finding a good naming.

Firestore data structure
https://www.youtube.com/watch?v=haMOUb3KVSo

Firestore Data Modeling I/O 2019 
https://www.youtube.com/watch?v=lW7DWV2jST0